plumbing/format/packfile/delta_index.go

```diff
@@ -32,19 +32,17 @@ func (idx *deltaIndex) findMatch(src, tgt []byte, tgtOffset int) (srcOffset, l i
 		return 0, -1
 	}
 
-	if len(tgt) >= tgtOffset+s && len(src) >= blksz {
-		h := hashBlock(tgt, tgtOffset)
-		tIdx := h & idx.mask
-		eIdx := idx.table[tIdx]
-		if eIdx != 0 {
-			srcOffset = idx.entries[eIdx]
-		} else {
-			return
-		}
-
-		l = matchLength(src, tgt, tgtOffset, srcOffset)
+	h := hashBlock(tgt, tgtOffset)
+	tIdx := h & idx.mask
+	eIdx := idx.table[tIdx]
+	if eIdx != 0 {
+		srcOffset = idx.entries[eIdx]
+	} else {
+		return
 	}
 
+	l = matchLength(src, tgt, tgtOffset, srcOffset)
+
```

Comment:
```suggestion
	h := hashBlock(tgt, tgtOffset)
	eIdx := idx.table[h & idx.mask]
	if eIdx == 0 {
		return
	} 
	
	srcOffset = idx.entries[eIdx]
	l = matchLength(src, tgt, tgtOffset, srcOffset)
```
I think applying early return could simplify the function a bit more.  How do you think @edigaryev?

Reply:
Sounds good, implemented in 932b08f13352fdcb9c193908bafac83e0dabaac0.